### PR TITLE
Manage Prow Trusted Cluster in Terraform

### PR DESCRIPTION
**Pull request by upodroid**
Started managing prow-tests in GKE.

I made a few modifications:
- new nodepool
- subscribed the cluster to a release channel (stable)
- nodepool now runs with a custom service account. This account has enough permissions to pull images, write logs, write metrics
- Start managing some IAM

/cc @kvmware @chizhg

### CODE REVIEW

**Comment by upodroid** [comment]
Looking good so far:

```
 REDACTED  MCW0CDP3YY  ~  Desktop  …  kn-test-infra  infra  gcp   manage-gke-in-tf  4⚑  $    terraform apply -target module.testsmodule.tests.module.project.module.project-factory.random_id.random_project_id_suffix: Refreshing state... [id=gPI]
module.tests.module.prow_trusted.random_string.cluster_service_account_suffix: Refreshing state... [id=awkz]
module.tests.google_service_account.flaky_test_reporter: Refreshing state... [id=projects/knative-tests/serviceAccounts/flaky-test-reporter@knative-tests.iam.gserviceaccount.com]
module.tests.module.project.module.project-factory.google_project.main: Refreshing state... [id=projects/knative-tests]
module.tests.google_service_account_iam_binding.flaky_test_reporter: Refreshing state... [id=projects/knative-tests/serviceAccounts/flaky-test-reporter@knative-tests.iam.gserviceaccount.com/roles/iam.workloadIdentityUser]
module.tests.module.project.module.project-factory.module.project_services.google_project_service.project_services["secretmanager.googleapis.com"]: Refreshing state... [id=knative-tests/secretmanager.googleapis.com]
module.tests.module.project.module.project-factory.module.project_services.google_project_service.project_services["compute.googleapis.com"]: Refreshing state... [id=knative-tests/compute.googleapis.com]
module.tests.google_service_account.gke_nodes: Refreshing state... [id=projects/knative-tests/serviceAccounts/gke-nodes@knative-tests.iam.gserviceaccount.com]
module.tests.module.iam.google_project_iam_binding.project_iam_authoritative["default--roles/storage.objectViewer"]: Refreshing state... [id=knative-tests/roles/storage.objectViewer]
module.tests.module.iam.google_project_iam_binding.project_iam_authoritative["default--roles/artifactregistry.reader"]: Refreshing state... [id=knative-tests/roles/artifactregistry.reader]
module.tests.module.iam.google_project_iam_binding.project_iam_authoritative["default--roles/stackdriver.resourceMetadata.writer"]: Refreshing state... [id=knative-tests/roles/stackdriver.resourceMetadata.writer]
module.tests.module.iam.google_project_iam_binding.project_iam_authoritative["default--roles/container.admin"]: Refreshing state... [id=knative-tests/roles/container.admin]
module.tests.module.iam.google_project_iam_binding.project_iam_authoritative["default--roles/logging.logWriter"]: Refreshing state... [id=knative-tests/roles/logging.logWriter]
module.tests.module.iam.google_project_iam_binding.project_iam_authoritative["default--roles/monitoring.metricWriter"]: Refreshing state... [id=knative-tests/roles/monitoring.metricWriter]
module.tests.module.iam.google_project_iam_binding.project_iam_authoritative["default--roles/cloudbuild.builds.editor"]: Refreshing state... [id=knative-tests/roles/cloudbuild.builds.editor]
module.tests.module.prow_trusted.random_shuffle.available_zones: Refreshing state... [id=-]
module.tests.module.prow_trusted.google_container_cluster.primary: Refreshing state... [id=projects/knative-tests/locations/us-central1-a/clusters/prow-trusted]
module.tests.module.prow_trusted.google_container_node_pool.pools["prod-v1"]: Refreshing state... [id=projects/knative-tests/locations/us-central1-a/clusters/prow-trusted/nodePools/prod-v1]
module.tests.module.prow_trusted.module.gcloud_delete_default_kube_dns_configmap.module.gcloud_kubectl.null_resource.module_depends_on[0]: Refreshing state... [id=5378125291546907061]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.
╷
│ Warning: Resource targeting is in effect
│ 
│ You are creating a plan with the -target option, which means that the result of this plan may not represent all of the changes requested
│ by the current configuration.
│ 
│ The -target option is not for routine use, and is provided only for exceptional situations such as recovering from errors or mistakes, or
│ when Terraform specifically suggests to use it as part of an error message.
╵
╷
│ Warning: Applied changes may be incomplete
│ 
│ The plan was created with the -target option in effect, so some changes requested in the configuration may have been ignored and the
│ output values may not be fully updated. Run the following command to verify that no other changes are pending:
│     terraform plan
│ 
│ Note that the -target option is not suitable for routine use, and is provided only for exceptional situations such as recovering from
│ errors or mistakes, or when Terraform specifically suggests to use it as part of an error message.
╵

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
```

File: infra/gcp/tests/main.tf [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  # Sane project defaults
+  default_service_account     = "keep"
+  disable_services_on_destroy = false
+  create_project_sa           = false
+  random_project_id           = false
+  auto_create_network         = true
+
+
+  activate_apis = [
```
Comment (chizhg) [inline infra/gcp/tests/main.tf:19]: Are these the only APIs needed to be enabled for knative-tests project?

File: infra/gcp/tests/gke.tf [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  subnetwork                 = "default"
+  ip_range_pods              = "gke-prow-trusted-pods-f579223d"
+  ip_range_services          = "gke-prow-trusted-services-f579223d"
+  http_load_balancing        = true
+  network_policy             = false
+  horizontal_pod_autoscaling = true
+  filestore_csi_driver       = false
+  create_service_account     = false
+
+  node_pools = [
```
Comment (chizhg) [inline infra/gcp/tests/gke.tf:21]: I saw you've already switched to the new node pool, which is nice! 

I have a general question about terraform, when we run `terraform apply` to reconcile the config changes like this, will there be downtime? Or does terraform always follow steps like https://cloud.google.com/kubernetes-engine/docs/tutorials/migrating-node-pool to ensure there is minimal impact?

Comment (upodroid) [reply infra/gcp/tests/gke.tf:21]: > I have a general question about terraform, when we run `terraform apply` to reconcile the config changes like this, will there be downtime? Or does terraform always follow steps like https://cloud.google.com/kubernetes-engine/docs/tutorials/migrating-node-pool to ensure there is minimal impact?

No. You need to create a new nodepool and then cordon & drain manually. For this particular cluster, it is not busy so making the new one and deleting the old one right away is fine. For Prow/Prow Build we will need to cordon + drain for much longer

Comment (chizhg) [reply infra/gcp/tests/gke.tf:21]: Got it, thanks for the explanation!

**Comment by chizhg** [comment]
/lgtm
/hold
Please feel free to unhold if the answer is "yes" to my other comment.

File: infra/gcp/tests/main.tf (hunk 1 above)
Comment (upodroid) [reply infra/gcp/tests/main.tf:19]: This is useful for new projects as you need to enable the APIs before you can create resources from a particular API. In our case, the project has been around for a long time with relevant APIs already enabled.

**Comment by upodroid** [comment]
/hold cancel

I'll work on the remaining clusters next. I'll cordon + drain the old nodepools gracefully as it is running workloads.

**Pull request merged**